library/ssl_tls.c

```diff
@@ -1100,6 +1100,9 @@ static int ssl_handshake_init(mbedtls_ssl_context *ssl)
     /* Initialize structures */
     mbedtls_ssl_session_init(ssl->session_negotiate);
     ssl_handshake_params_init(ssl->handshake);
+#if defined(MBEDTLS_SSL_EARLY_DATA) && defined(MBEDTLS_SSL_SRV_C)
+    ssl->session_negotiate->max_early_data_size = ssl->conf->max_early_data_size;
+#endif
```

Comment:
We do not need that I'd say, just use `ssl->conf->max_early_data_size` where needed.

Reply:
Yes. That's not scope of this PR. 

It is target to resolve early data size in 1st flight, it is not current target.

Reply:
Ticket includes max_early_data_size, when issue new ticket, we should put it in ticket and early_data extension, then both client and server will got same value.